src/pfe/file-watcher/scripts/root-watcher.sh

```diff
@@ -91,6 +95,7 @@ if [ "$IN_K8" == "true" ]; then
 	echo "Initializing a custom helm tiller"
 	helm init --upgrade --service-account $SERVICE_ACCOUNT_NAME
 
+	# TO BE REMOVED AS PART OF CODEWIND ISSUE https://github.com/eclipse/codewind/issues/665 WHEN APIS ARE READY TO BE CONSUMED
```

Comment:
What happens when PFE restarts and the .dockerconfigjson under `/root/.docker/` is lost?

I'm wondering if we instead want to adapt this for new architecture and have PFE pull down the config.json from the secret when it starts up, if a secret already exists.

Reply:
so we want to do this in root-watcher.sh?

Reply:
We could adapt the existing code to pull down the .dockerconfigjson with kubectl _or_ handle it elsewhere. It doesn't matter where specifically, so long as the .dockerconfigjson isn't lost when PFE restarts

Reply:
So the Q is how does the logic decide if it was a PFE restart? Since when PFE starts there never is a `/roott/.docker/config.json` until the API is called

Reply:
Well, in `root-watcher.sh` (the closest to a Turbine entrypoint there is), we could check for a registry secret labeled with the instance's workspace ID. If it exists, we pull down the `.dockerconfigjson` file under `/root/.docker/`. If no secret exists, we create an empty `/root/.docker`

Reply:
yeah makes sense, tho its gonna involve using kubectl and base64/openssl commands

Reply:
updated behind a flag for safe keeping